### PR TITLE
EENUGW

### PR DESCRIPTION
**Pull request by Dragonnhun**

### CODE REVIEW

File: EENUGW/beadando/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+data "google_project" "project" {
```
Comment (s1mark) [inline EENUGW/beadando/outputs.tf:1]: please put the data types in the main.tf
the outputs should only contain output types

**Review by s1mark — CHANGES_REQUESTED** [review]
Try running terraform plan

**Comment by Dragonnhun** [comment]
<details>
<summary>terraform plan</summary>

```
PS C:\EKCU\Cloud\cloud\EENUGW\beadando> terraform plan
data.google_compute_disk.disk: Reading...
data.google_project.project: Reading...
data.google_compute_disk.disk: Read complete after 0s
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_compute_disk.disk will be created
  + resource "google_compute_disk" "disk" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "eenugw"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # google_compute_instance.instance will be created
  + resource "google_compute_instance" "instance" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done >> ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "eenugw"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = (known after apply)
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # google_service_account.service_account will be created
  + resource "google_service_account" "service_account" {
      + account_id   = "eenugw"
      + disabled     = false
      + display_name = "EENUGW"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + project_id     = "projects/ekke-cloud"
  + project_name   = "ekke-cloud"
  + project_number = "92227481840"
```

</details>
<details>
<summary>terraform apply</summary>

```
PS C:\EKCU\Cloud\cloud\EENUGW\beadando> terraform apply
data.google_project.project: Reading...
data.google_compute_disk.disk: Reading...
data.google_compute_disk.disk: Read complete after 0s
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_compute_disk.disk will be created
  + resource "google_compute_disk" "disk" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "eenugw"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # google_compute_instance.instance will be created
  + resource "google_compute_instance" "instance" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done >> ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "eenugw"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = (known after apply)
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # google_service_account.service_account will be created
  + resource "google_service_account" "service_account" {
      + account_id   = "eenugw"
      + disabled     = false
      + display_name = "EENUGW"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + project_id     = "projects/ekke-cloud"
  + project_name   = "ekke-cloud"
  + project_number = "92227481840"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_service_account.service_account: Creating...
google_compute_disk.disk: Creating...
google_service_account.service_account: Creation complete after 1s [id=projects/ekke-cloud/serviceAccounts/eenugw@ekke-cloud.iam.gserviceaccount.com]
google_compute_disk.disk: Creation complete after 2s [id=projects/ekke-cloud/zones/europe-central2-a/disks/eenugw]
google_compute_instance.instance: Creating...
google_compute_instance.instance: Still creating... [10s elapsed]
google_compute_instance.instance: Creation complete after 13s [id=projects/ekke-cloud/zones/europe-central2-a/instances/eenugw]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

project_id = "projects/ekke-cloud"
project_name = "ekke-cloud"
project_number = "92227481840"
```
</details>

Note: the outputs related to "google_compute_disk" only appear after using terraform plan (and terraform apply) again.

<details>
<summary>terraform plan and terraform apply</summary>

```
PS C:\EKCU\Cloud\cloud\EENUGW\beadando> terraform plan
data.google_project.project: Reading...
data.google_compute_disk.disk: Reading...
google_service_account.service_account: Refreshing state... [id=projects/ekke-cloud/serviceAccounts/eenugw@ekke-cloud.iam.gserviceaccount.com]
google_compute_disk.disk: Refreshing state... [id=projects/ekke-cloud/zones/europe-central2-a/disks/eenugw]
data.google_compute_disk.disk: Read complete after 0s [id=projects/ekke-cloud/zones/europe-central2-a/disks/eenugw]
google_compute_instance.instance: Refreshing state... [id=projects/ekke-cloud/zones/europe-central2-a/instances/eenugw]
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Changes to Outputs:
  + disk_image     = "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-11-bullseye-v20220719"
  + disk_self_link = "https://www.googleapis.com/compute/v1/projects/ekke-cloud/zones/europe-central2-a/disks/eenugw"
  + disk_size      = 10
  + disk_type      = "pd-standard"

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

PS C:\EKCU\Cloud\cloud\EENUGW\beadando> terraform apply
data.google_compute_disk.disk: Reading...
data.google_project.project: Reading...
google_service_account.service_account: Refreshing state... [id=projects/ekke-cloud/serviceAccounts/eenugw@ekke-cloud.iam.gserviceaccount.com]
google_compute_disk.disk: Refreshing state... [id=projects/ekke-cloud/zones/europe-central2-a/disks/eenugw]
data.google_compute_disk.disk: Read complete after 0s [id=projects/ekke-cloud/zones/europe-central2-a/disks/eenugw]
google_compute_instance.instance: Refreshing state... [id=projects/ekke-cloud/zones/europe-central2-a/instances/eenugw]
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Changes to Outputs:
  + disk_image     = "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-11-bullseye-v20220719"
  + disk_self_link = "https://www.googleapis.com/compute/v1/projects/ekke-cloud/zones/europe-central2-a/disks/eenugw"
  + disk_size      = 10
  + disk_type      = "pd-standard"

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

disk_image = "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-11-bullseye-v20220719"
disk_self_link = "https://www.googleapis.com/compute/v1/projects/ekke-cloud/zones/europe-central2-a/disks/eenugw"
disk_size = 10
disk_type = "pd-standard"
project_id = "projects/ekke-cloud"
project_name = "ekke-cloud"
project_number = "92227481840"
```

</details

**Comment by s1mark** [comment]
Please share the plan and apply results in the comments

**Comment by Dragonnhun** [comment]
You can click on these dropdowns and the output will appear:
kép

**Comment by s1mark** [comment]
> Please share the plan and apply results in the comments



> You can click on these dropdowns and the output will appear: kép

ouch didnt see :)

**Review by s1mark — APPROVED** [review]

**Comment by s1mark** [comment]
Good job

**Comment by Dragonnhun** [comment]
Yeah it wasn't obvious, sorry.

Thank you!

**Pull request merged**